### ISSUE TITLE
See if using ShadowRoot.delegatesFocus allows us to call .focus on any focusable calcite component

### ISSUE THREAD

**Issue by eriklharper** [Esri/calcite-components#3626]
### Background 

Focusable calcite components have been implementing a `setFocus` method as a recommended convention from Ionic, but the [`HTMLElement.focus()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus) method would be nicer to be able to use since it is common to all HTMLElements.  This issue will explore whether using [`ShadowRoot.delegatesFocus`](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/delegatesFocus) will allow any focusable calcite component to be able to rely on the built-in focus method.

If this works, even though this feature isn't fully supported in all the browser versions Calcite officially supports, it is likely we'll be able to implement this in addition to keeping the `setFocus` implementation around, at least until it is no longer needed.

https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/delegatesFocus

### Desired Outcome 

Build a prototype example showcasing the ability to call `focus()` on a focusable calcite component using `ShadowRoot.delegatesFocus`.

### Additional Resources

* https://developers.google.com/web/fundamentals/web-components/shadowdom#focus
* https://github.com/calebdwilliams/delegates-focus-polyfill

**Comment by benelan** [comment]
You can use `focus()` on component's with `delegatesFocus` enabled like this:

```js
(async () => {
  await customElements.whenDefined("calcite-date-picker");
  const el = await document
    .querySelector("calcite-date-picker")
    .componentOnReady();
  requestAnimationFrame(() => el.setFocus());
})();
```

After #6156 it will be simplified to

```js
 customElements
    .whenDefined("calcite-date-picker")
    .then(() => document.querySelector("calcite-date-picker").focus());
```

Not all of the components have `delegatesFocus` enabled right now, see https://github.com/Esri/calcite-components/issues/5147#issuecomment-1349616672 for more info. But in terms of the research for this issue I think that's done and confirmed that it will work. Should we close this? @driskull @eriklharper @geospatialem

**Comment by driskull** [comment]
Yeah lets close this 👍 

I still think we will keep our `setFocus` methods in the near future because inside of our setFocus methods we can:

- Wait for `componentOnReady()` since our setFocus methods are async and return promises
- Wait for `requestAnimationFrame()`

I think those are the only two benefits of having our own focus methods. Otherwise, as noted by @benelan, the user has to  wait for `componentOnReady` and `requestAnimationFrame()`
